src/ERC7540AsyncDepositExample.sol

```diff
@@ -1,8 +1,15 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.15;
 
-import "solmate/mixins/ERC4626.sol";
-import "solmate/auth/Owned.sol";
+import {IERC7540Deposit} from "src/interfaces/IERC7540.sol";
+import {ERC4626} from "solmate/mixins/ERC4626.sol";
+import {IERC4626} from "src/interfaces/IERC4626.sol";
+import {IERC165} from "src/interfaces/IERC7575.sol";
+import {SafeTransferLib} from "src/libraries/SafeTransferLib.sol";
```

Comment:
Since we already have `solmate` imported, I would use https://github.com/transmissions11/solmate/blob/main/src/utils/SafeTransferLib.sol